### PR TITLE
lab4

### PR DESCRIPTION
**Pull request by asapmandarinka**

### CODE REVIEW

File: city.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        self.balance_per_day[motif] += amount
+    def finalize_balance_per_day(self) -> None:
+        for motif in self.balance_per_day:
+            self.balance[motif] += self.balance_per_day[motif]
+            self.balance_per_day[motif] = 0
+        if not self.full:
+            for motif in self.balance_per_day:
+                if self.balance[motif] == 0:
+                    return
+            self.full = True
```
Comment (dmitry9933) [inline city.py:39]: add padding

File: map.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    def simulate_euro_diffusion(self) -> None:
+        if len(self.countries) == 1:
+            self.countries[0].day_of_full = 0
+            return
+
+        full = False
+        day = 1
+        while not full:
+
```
Comment (dmitry9933) [inline map.py]: add 
for x in range(self._grid_offset + 1):
                for y in range(self._grid_offset + 1):
                    if self.grid[x][y] is not None:
                        city = self.grid[x][y]
                        city.transfer_to_neighbours()

File: country.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        self.full = True
+        self.day_of_full = day
+
+    def has_foreign_neighbours(self) -> bool:
+        for city in self.cities:
+            for neighbour in city.neighbours:
+                if neighbour.country_name != self.name:
+                    return True
+
+    def only_country_mode(self) -> None
```
Comment (dmitry9933) [inline country.py]: add ":" after None, error

**Comment by asapmandarinka** [comment]
done

**Pull request merged**